x-pack/plugins/cases/public/containers/use_get_feature_ids.tsx

```diff
@@ -6,28 +6,66 @@
  */
 
 import { useQuery } from '@tanstack/react-query';
-import type { ValidFeatureId } from '@kbn/rule-data-utils';
+import { isValidFeatureId } from '@kbn/rule-data-utils';
+import { useEffect, useMemo, useState } from 'react';
 import type { ServerError } from '../types';
 import { useCasesToast } from '../common/use_cases_toast';
 import * as i18n from './translations';
+import type { FeatureIdsResponse } from './api';
 import { getFeatureIds } from './api';
 import { casesQueriesKeys } from './constants';
 
-export const useGetFeatureIds = (alertRegistrationContexts: string[]) => {
+export const useGetFeatureIds = (
+  alertIds: string[],
+  alertIdsQuery: {
+    ids: {
+      values: string[];
+    };
+  },
+  enabled: boolean
+) => {
   const { showErrorToast } = useCasesToast();
-
-  return useQuery<ValidFeatureId[], ServerError>(
-    casesQueriesKeys.alertFeatureIds(alertRegistrationContexts),
+  const [featureIds, setFeatureIds] = useState<string[]>([]);
+  const { data, isSuccess, isInitialLoading, isLoading } = useQuery<
+    FeatureIdsResponse,
+    ServerError
+  >(
+    casesQueriesKeys.alertFeatureIds(alertIds),
     ({ signal }) => {
-      const query = { registrationContext: alertRegistrationContexts };
-      return getFeatureIds({ query, signal });
+      return getFeatureIds({ query: alertIdsQuery, signal });
     },
     {
+      enabled,
       onError: (error: ServerError) => {
         showErrorToast(error, { title: i18n.ERROR_TITLE });
       },
     }
   );
+
+  useEffect(() => {
```

Comment:
If this `useEffect` is to transform the data you can use the `select` property of `react-query`. You can see an example here: `x-pack/plugins/cases/public/containers/user_profiles/use_bulk_get_user_profiles.ts`.